scipy/linalg/tests/test_decomp.py

```diff
@@ -2146,6 +2146,39 @@ def test_diff_input_types(self):
         ret = ordqz(self.B2, self.A1, sort='lhp')
         self.check(self.B2, self.A1, 'lhp', *ret)
 
+class TestOrdQZWorkspaceSize(TestCase):
+
+    def setUp(self):
+        seed(12345)
+
+    def test_decompose(self):
+
+        import numpy.random
+        from scipy.linalg import ordqz
+
+        N = 202
+
+        # raises error if lwork parameter to dtrsen is too small
+        for ddtype in [np.float32, np.float64]:
+            A = numpy.random.random((N,N)).astype(ddtype)
+            B = numpy.random.random((N,N)).astype(ddtype)
+            # sort = lambda alphar, alphai, beta: alphar**2 + alphai**2< beta**2
+            sort = lambda alpha, beta: alpha < beta
+            [S,T,alpha,beta,U,V] = ordqz(A,B,sort=sort, output='real')
+
+        for ddtype in [np.complex, np.complex64]:
+            A = numpy.random.random((N,N)).astype(ddtype)
+            B = numpy.random.random((N,N)).astype(ddtype)
+            sort = lambda alpha, beta: alpha < beta
+            [S,T,alpha,beta,U,V] = ordqz(A,B,sort=sort, output='complex')
+
+        # # segfaults if lwork parameter to dtrsen is too small
+        # for ddtype in [np.float32, np.float64, np.complex, np.complex64]:
+        #     A = numpy.random.random((N,N)).astype(ddtype)
+        #     B = numpy.random.random((N,N)).astype(ddtype)
+        #     [S,T,alpha,beta,U,V] = ordqz(A,B,sort='ouc')
```

Comment:
On my machine, float32 segfaults but float64 does not. Maybe the right thing to do is to separate this into a test method of its own and decorate it with `@slow`.